themes/default/content/blog/functions-accept-outputs/index.md

```diff
@@ -0,0 +1,391 @@
+	ApplyT(func(id string) *ecr.GetCredentialsResult {
+		creds, err := ecr.GetCredentials(ctx,
+			&ecr.GetCredentialsArgs{RegistryId: id})
+		if err != nil {
+			panic(err)
+		}
+		return creds
+	}).(ecr.GetCredentialsResultOutput)
+```
+
+{{% /choosable %}}
+{{% choosable language csharp %}}
+
+```csharp
+var creds = repo.Id.Apply(repoId =>
+                          GetCredentials.InvokeAsync(new GetCredentialsArgs
+                          {
+                              RegistryId = repoId
+                          }));
+```
+
+{{% /choosable %}}
+
+## More examples
+
+The above example is one of many practical situations where mixing
+function calls and resources benefits from the new form. To find out
+more, check out the following updated Pulumi examples:
+
+- [azure-py-appservice example](https://github.com/pulumi/examples/tree/master/azure-py-appservice)
+- [aws-ts-appsync example](https://github.com/pulumi/examples/tree/master/aws-ts-appsync)
+- [azure-cs-aks-cosmos-helm](https://github.com/pulumi/examples/tree/master/azure-cs-aks-cosmos-helm)
+
+## Compatibility
+
+To keep existing Pulumi programs working without changes, the function
+forms are added as separate functions or methods in each
+Pulumi-supported language following a simple naming convention. To
+illustrate with the `getCredentials` function:
+
+| Language   | Existing non-Output form   | New Output form                      |
+|------------|----------------------------|--------------------------------------|
+| TypeScript | aws.ecr.getCredentials     | aws.ecr.getCredentials**Output**     |
+| Python     | aws.ecr.get\_credentials   | aws.ecr.get\_credentials\_**output** |
+| Go         | ecr.getCredentials         | ecr.getCredentials**Output**         |
+| C#         | GetCredentials.InvokeAsync | GetCredentials.**Invoke**            |
+
+Note that there are cases where the existing non-`Output` form may
+still be the right choice. For example, retrieving the default VPC in
+Python utilizing the existing form is simpler as it returns a result
+that can be immediately inspected:
+
+```python
+default_vpc = aws.ec2.get_vpc(default=True)
+print(default_vpc.id)
+```
+
+Prefer the new Output form when passing resource outputs to a function
+or else using the outputs of the function as inputs to resources. You
+may still want to use the existing non-Output form if you are using
+the outputs of the function to inform control flow (`if` conditionals
+or `for` loops).
+
+## Get started
+
+To use Output-versioned functions, please upgrade your install of
+Pulumi to at least 3.17.1 and upgrade your providers to the latest
+available version. Example compatible versions for major Pulumi
+providers:
+
+| Provider             | Version |
+|----------------------|---------|
+| pulumi-aws           | 4.27.0  |
+| pulumi-azure-native  | 1.45.0  |
+| pulumi-google-native | 0.8.0   |
+| pulumi-azure         | 4.26.0  |
+| pulumi-gcp           | 5.26.0  |
```

Comment:
This all looks really good. 1 nit:
There needs to be an extra space to accommodate the tags at the bottom (example below).
```suggestion
| pulumi-gcp           | 5.26.0  |


```
<img width="753" alt="Screen Shot 2021-11-12 at 3 05 34 PM" src="https://user-images.githubusercontent.com/22222529/141593571-65f12d93-8fd9-46e9-b772-ff20f5c09d44.png">